packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -657,6 +662,15 @@ class _DatePickerDialogState extends State<_DatePickerDialog> {
 
   @override
   Widget build(BuildContext context) {
+    switch (Theme.of(context).platform) {
+      case TargetPlatform.android:
+      case TargetPlatform.fuchsia:
+        _vibrate = true;
+        break;
+      case TargetPlatform.iOS:
+        _vibrate = false;
+        break;
+    }
```

Comment:
Rather than precomputing this information in build, I'd recommend computing it on-the-fly when we try to vibrate.  A State object has a `context` property that you can feed to `Theme.of` whenever you like.